odoo/addons/base/models/res_partner.py

```diff
@@ -213,7 +213,7 @@ def default_get(self, default_fields):
     company_type = fields.Selection(string='Company Type',
         selection=[('person', 'Individual'), ('company', 'Company')],
         compute='_compute_company_type', inverse='_write_company_type')
-    company_id = fields.Many2one('res.company', 'Company', index=True)
+    company_id = fields.Many2one('res.company', 'Company', index=True, domain=lambda self: [('id', 'in', self.env.companies.ids)])
```

Comment:
As we consider that base usage of partners is to be cross companies, shouldn't we simply keep the user_id reset and let other fields as they are ?